lib/datasource/common.ts

```diff
@@ -50,3 +52,12 @@ export interface Datasource {
   getPreset?(packageName: string, presetName?: string): Promise<Preset>;
   getPkgReleases(config: PkgReleaseConfig): Promise<ReleaseResult | null>;
 }
+
+export class DatasourceError extends Error {
+  err: Error;
+
+  constructor(err: Error) {
+    super(DATASOURCE_FAILURE);
+    this.err = err;
+  }
+}
```

Comment:
```suggestion 
export class DatasourceError extends Error {
  err: Error;
  name: string;

  constructor(name: string, err: Error) {
    super(DATASOURCE_FAILURE);
    this.err = err;
    this.name = name;
  }
```

Reply:
I would prefer to reuse `datasource`, plus also I thought `name` had a special meaning in errors